c10/cuda/CUDACachingAllocator.cpp

```
@@ -180,6 +185,43 @@ struct AllocParams {
 
 } // namespace
 
+
+class CachingAllocatorConfig {
+public:
+  size_t max_split_size;  // largest "large" alloc, "oversize" is not splitable
+
+  CachingAllocatorConfig()
+  : max_split_size(SIZE_MAX)
+  {
+    const char *val = getenv("PYTORCH_CUDA_ALLOC_CONF");
+    if (val != NULL) {
+      std::string config(val);
+      size_t pos = 0;
+
+      while (pos < config.length()) {
+        size_t start = pos;
+        size_t end = config.find(',', start);
+        std::string line = config.substr(start, end - start);
+        size_t seperator = config.find(':', start);
+        if (seperator < end) {
+          std::string key = config.substr(start, seperator - start);
+          std::string value = config.substr(seperator+1, end - seperator - 1);
+          // Handle each config option
+          if (key.compare("max_split_size_mb") == 0) {
+              int val = stoi(value);
+              val = val < 20 ? 20 : val;
+              val = val > (size_t(SIZE_MAX) / (1024*1024)) ? (size_t(SIZE_MAX) / (1024*1024)) : val;
+              max_split_size = val * 1024 * 1024;
+          }
```

so unrecognized options are silently ignored? That doesn't seem very good